compiler/rustc_borrowck/src/diagnostics/explain_borrow.rs

```diff
@@ -390,6 +428,38 @@ impl<'tcx> BorrowExplanation<'tcx> {
     }
 }
 
+fn suggest_rewrite_if_let<'tcx>(
+    expr: &hir::Expr<'tcx>,
+    pat: &str,
+    init: &hir::Expr<'tcx>,
+    conseq: &hir::Expr<'tcx>,
+    alt: Option<&hir::Expr<'tcx>>,
+    err: &mut Diag<'_>,
+) {
+    err.span_note(
+        conseq.span.shrink_to_hi(),
+        "lifetime for temporaries generated in `if let`s have been shorted in Edition 2024",
```

Comment:
Before I forget: typo.
```suggestion
        "lifetime for temporaries generated in `if let`s have been shortened in Edition 2024",
```

Reply:
Fixed